rclcpp/include/rclcpp/client.hpp

```diff
@@ -44,15 +46,21 @@ class ClientBase
 public:
   RCLCPP_MAKE_SHARED_DEFINITIONS(ClientBase);
 
-  ClientBase(rmw_client_t * client_handle, const std::string & service_name)
-  : client_handle_(client_handle), service_name_(service_name)
+  ClientBase(
+    std::shared_ptr<rmw_node_t> node_handle,
+    rmw_client_t * client_handle,
+    const std::string & service_name)
+  : node_handle_(node_handle), client_handle_(client_handle), service_name_(service_name)
   {}
 
   ~ClientBase()
   {
-    if (client_handle_ != nullptr) {
-      rmw_destroy_client(client_handle_);
-      client_handle_ = nullptr;
+    if (client_handle_) {
+      if (rmw_destroy_client(client_handle_) == RMW_RET_ERROR) {
+        std::cerr << "Error in destruction of rmw client handle: " <<
+        (rmw_get_error_string() ? rmw_get_error_string() : "") <<
+          std::endl;
```

Comment:
That's what `uncrustify` does this to it... I'm pretty nonplussed about it's ability to handle some of these cases, but I don't have time to look into why it does that or come up with another solution. I guess I can add the comments around it to disable the check.